controllers/azuremanagedcontrolplane_controller.go

```diff
@@ -98,7 +98,7 @@ func (amcpr *AzureManagedControlPlaneReconciler) SetupWithManager(ctx context.Co
 	// Add a watch on clusterv1.Cluster object for unpause & ready notifications.
 	if err = c.Watch(
 		&source.Kind{Type: &clusterv1.Cluster{}},
-		handler.EnqueueRequestsFromMapFunc(util.ClusterToInfrastructureMapFunc(ctx, infrav1.GroupVersion.WithKind("AzureManagedControlPlane"), mgr.GetClient(), &infrav1.AzureManagedControlPlane{})),
+		handler.EnqueueRequestsFromMapFunc(amcpr.ClusterToAzureManagedControlPlane),
 		predicates.ClusterUnpausedAndInfrastructureReady(log),
 		predicates.ResourceNotPausedAndHasFilterLabel(log, amcpr.WatchFilterValue),
 	); err != nil {
```

Comment:
Sorry I am late getting to this PR, but I am curious if we could achieve reconciling `AzureManagedControlPlane` by updating the predicate `predicates.ClusterUnpausedAndInfrastructureReady(log)` to `predicates.ClusterUnpaused(log)`?